### ISSUE TITLE
pure numbers as link targets produce warning

### ISSUE THREAD

**Issue by rpuntaie** [sphinx-doc/sphinx#4434]
```
.. _`03234`: 

|03234|

.. |03234| replace:: :ref:`Text <03234>`
```

warns about 

```
index.rst:6: WARNING: undefined label: 03234 (if the link has no caption the label must precede a section header)
```

but, if prefixing each 03234 by `a` character there is no such warning.

I don't see a reason why numbers alone should not work the same in this context.

`rst2html.py` does not produce a warning.

**Comment by tk0miya** [comment]
Thank you for reporting.

At present, Sphinx expects labels are not digit.
https://github.com/sphinx-doc/sphinx/blob/a9f949e8580fbd685b8b334a875852f2ad65de8a/sphinx/domains/std.py#L610-L614

Note: this code was added at 910e488e39.

**Comment by tk0miya** [comment]
Fixed by #4461 
Thank you for reporting!
